chain/indexer/src/streamer/streamer.rs

```diff
@@ -194,11 +197,30 @@ async fn fetch_chunks(
 pub(crate) async fn start(
     view_client: Addr<near_client::ViewClientActor>,
     client: Addr<near_client::ClientActor>,
+    indexer_config: IndexerConfig,
     mut blocks_sink: mpsc::Sender<BlockResponse>,
 ) {
     info!(target: INDEXER, "Starting Streamer...");
+    let mut indexer_db_path = neard::get_store_path(&indexer_config.home_dir);
+    indexer_db_path.push_str("/indexer");
+
+    let db = DB::open_default(indexer_db_path).unwrap();
     let mut outcomes_to_get = Vec::<types::TransactionOrReceiptId>::new();
-    let mut last_synced_block_height: types::BlockHeight = 0;
+    let mut last_synced_block_height: types::BlockHeight = match indexer_config.sync_mode {
+        crate::SyncModeEnum::FromInterruption => match db.get(b"last_synced_block_height") {
+            Ok(Some(value)) => String::from_utf8(value).unwrap().parse::<u64>().unwrap(),
+            Ok(None) => 0,
+            Err(_) => {
+                db.put(b"last_synced_block_height", b"0").unwrap();
+                0
+            }
+        },
+        crate::SyncModeEnum::LatestSynced => 0, // let streamer to handle from real-time
+        crate::SyncModeEnum::BlockHeight(height) => height - 1, // provided block inclusively
```

Comment:
Having `BlockHeight(0)` will cause overflow and streamer will never send a block since the `latest_synced_block_height` will be u64::MAX, so you need to handle this case:

```suggestion
        crate::SyncModeEnum::BlockHeight(height) => height.checked_sub(1).unwrap_or(0), // provided block inclusively
```